script/upload

```diff
@@ -0,0 +1,262 @@
+#!/bin/sh
+
+set -e
+
+REPO="git-lfs/git-lfs"
+WORKDIR="$(mktemp -d)"
+
+trap 'rm -fr "$WORKDIR"' EXIT
+
+say () {
+  [ -n "$QUIET" ] && return
+  local format="$1"
+  shift
+  printf "$format\n" "$@" >&2
+}
+
+abort () {
+  local format="$1"
+  shift
+  printf "$format\n" "$@" >&2
+  exit 2
+}
+
+curl () {
+  command curl -nfSs "$@"
+}
+
+categorize_os () {
+  local os="$1"
+
+  if [ "$os" = "freebsd" ]
+  then
+    echo FreeBSD
+  else
+    ruby -e 'puts ARGV[0].capitalize' "$os"
+  fi
+}
+
+categorize_arch () {
+  local arch="$1"
+
+  ruby -e 'puts ARGV[0].upcase' "$arch"
+}
+
+# Categorize a release asset and print its human readable name to standard
+# output.
+categorize_asset () {
+  local file="$1"
+  local os=$(echo "$file" | sed -e 's/^git-lfs-//' -e 's/[-.].*$//')
+  local arch=$(echo "$file" | ruby -pe '$_.gsub!(/\Agit-lfs-[^-]+-([^-]+)[-.].*/, "\\1")')
+
+  if ruby -e 'exit 1 unless /\Agit-lfs-v\d+\.\d+\.\d+\.tar\.gz\z/ =~ ARGV[0]' "$file"
```

Comment:
And the same here, but perhaps with a `case ... esac` statement?

Reply:
This does indeed read better as a case statement. I've update another function to use that approach as well.